src/driver/compile.c

```diff
@@ -399,14 +418,9 @@ static int yfc_validate_compile(
     if (retval)
         return retval;
 
-    if (yf_ensure_entry_point(pdata)) {
-        /* TODO - a more helpful error messsage */
-        YF_PRINT_ERROR("Expected exactly 1 'main' function.");
-        return 1;
-    }
-
-    if (adata->stage >= YF_COMPILE_CODEGEN)
+    if (adata->stage >= YF_COMPILE_CODEGENONLY) {
         retval = yf_backend_generate_code(adata);
+    }
```

Comment:
I think you want to add the check inside the if before generating code like
```
if (adata->stage == YF_COMPILE_FULL && yf_ensure_entry_point(pdata))
    return 1;
```

Reply:
Done.